lib/handlers/health/apps.js

```diff
@@ -0,0 +1,130 @@
+var request = require('request');
+var async = require('async');
+var fhconfig = require('fh-config');
+var logger = require('../../util/logger').getLogger();
+
+var dfc;
+
+// Default timeout: 15 seconds
+var DEFAULT_TIMEOUT = 15*1000;
+// URI called to check if the app is reachable
+var DEFAULT_PING_URI = '/sys/info/ping';
+
+if (!fhconfig.value('openshift3')) {
+  var fhdfc = require('fh-dfc');
+
+  dfc = fhdfc(fhconfig.value('fhdfc'));
+}
+
+/**
+ * Utility function to build a result object
+ * @param status result status
+ * @param error optional error message
+ * @returns {{status: *, error: *}}
+ */
+function result(status, error) {
+  return {
+    status: status,
+    error: error
+  };
+}
+
+/**
+ *
+ * @param dynos The list of runnning dynos
+ * @param callback callback used to return the result of the operation. It must be in the format
+ * cb(err, appData) where appData will be a map with this structure:
+ * { appId: { dyno: dynoName, url: appUrl } }
+ */
+function getAppsUrl(dynos, callback) {
+  var res = {};
+
+  async.each(dynos, function (dyno, cb) {
+    dfc.apps.list(dyno.dyno, 'detailed', function(err, apps) {
+      async.each(apps, function (app, cb1) {
+
+        if (app.state === 'RUNNING') {
+          dfc.url([dyno.dyno, app.app], function (err, url) {
+            res[app.app] = {'dyno': dyno.dyno, 'url': url};
+            cb1();
+          });
+        } else {
+          cb1();
+        }
+      }, function (err) {
+        cb(err);
+      });
+    })
+  }, function (err) {
+    callback(err, res);
+  });
+}
+
+/**
+ * Checks that the health is ok
+ * @param appData App data in the format {url: appUrl, dyno: dyno}
+ * @param appName Name of the app
+ * @param cb
+ */
+function checkAppHealth(appData, appName, cb) {
+  request({
+    baseUrl: appData.url,
+    uri: DEFAULT_PING_URI,
+    timeout: DEFAULT_TIMEOUT
+  }, function(err, response, body) {
+
+    if (err) {
+      logger.error('Error invoking health check endpoint', {appName: appName, appData: appData, uri: DEFAULT_PING_URI, err: err});
+      return cb(result('error', 'Error invoking health check endpoint'));
+    }
+
+    if (response.statusCode !== 200) {
+      logger.error('Error invoking endpoint', {appName: appName, appData: appData, err: response.statusMessage, code: response.statusCode});
+      return cb(result('error', response.statusMessage)); // TODO: send data about the failing app...
+    }
+
+    var res = JSON.parse(body);
+
+    if (res === 'OK') {
```

Comment:
I think we should extract all constant error codes (`OK` and `error`) to constants at the top of the file.